internal/server/heathcheck_server.go

```diff
@@ -48,26 +49,114 @@ func createReadinessChecker(serveOpts *ServeOptions) http.Handler {
 		Timeout: 5 * time.Second,
 	}
 
-	readinessChecker := health.NewChecker(
-		health.WithCheck(health.Check{
+	// Start with the MCP server check
+	checks := []health.Check{
+		{
 			Name: "mcp-server",
 			Check: func(ctx context.Context) error {
 				// Check connectivity to the MCP server using an MCP client.
 				return checkMCPServer(ctx, serveOpts)
 			},
-		}),
-		health.WithCheck(health.Check{
-			Name: "api-server",
-			Check: func(ctx context.Context) error {
-				// Check HTTP connectivity to the API server.
-				return checkAPIServerConnectivity(ctx, serveOpts, httpClient)
-			},
-		}),
-	)
+		},
+	}
+
+	// Add individual health checks for each OAS path
+	checks = append(checks, createOASPathHealthChecks(serveOpts, httpClient)...)
+
+	// Build the checker options
+	options := []health.CheckerOption{}
+	for _, check := range checks {
+		options = append(options, health.WithCheck(check))
+	}
+
+	readinessChecker := health.NewChecker(options...)
 
 	return health.NewHandler(readinessChecker)
 }
 
+// createOASPathHealthChecks creates individual health checks for each OAS path.
+func createOASPathHealthChecks(serveOpts *ServeOptions, httpClient *http.Client) []health.Check {
+	var checks []health.Check
+
+	// If we have explicit OAS paths, create a health check for each one
+	// assuming "<base path> + /api/healthz"
+	if len(serveOpts.OASPath) > 0 {
+		for _, oasPath := range serveOpts.OASPath {
+			check, err := createSingleOASHealthCheck(oasPath, serveOpts, httpClient)
+			if err != nil {
+				continue
+			}
+
+			checks = append(checks, check)
+		}
+		// If no explicit OAS paths but we have a TrentoURL, create checks for autodiscovery paths
+		// also assuming "<base path> + /api/healthz"
+	} else if serveOpts.TrentoURL != "" {
+		for _, autoPath := range serveOpts.AutodiscoveryPaths {
+			fullOASPath := strings.TrimRight(serveOpts.TrentoURL, "/") + autoPath
+
+			check, err := createSingleOASHealthCheck(fullOASPath, serveOpts, httpClient)
+			if err != nil {
+				continue
+			}
+
+			checks = append(checks, check)
+		}
+		// at this point, TrentoURL should be populated even with the fallback demo instance;
+		// if not, just return empty checks
+	}
+
+	return checks
+}
+
+// createSingleOASHealthCheck creates a single health check for an OAS path.
+func createSingleOASHealthCheck(
+	oasPath string,
+	serveOpts *ServeOptions,
+	httpClient *http.Client,
+) (health.Check, error) {
+	// Parse the OAS path once to extract information for the health check
+	parsedURL, err := url.Parse(oasPath)
+	if err != nil {
+		return health.Check{}, fmt.Errorf("failed to parse OAS path %s: %w", oasPath, err)
+	}
+
+	// Validate that the URL has a host (required for health checks)
+	if parsedURL.Host == "" {
+		return health.Check{}, fmt.Errorf("failed to extract host from OAS path %s", oasPath)
+	}
+
+	checkNameTpl := "%s-api"
```

Comment:
This is not going to work.
We are not adding the `/wanda` prefix in local deployments. Both subpaths for `web` and `wanda` will be `/api/all/openapi`, the difference being to which service we point.
I don't remember what for is used the `checkName` but I'm afraid we will need to have the whole OAS url in there.
And the same to create the `healthPath`. We cannot infer the the `/wanda` prefix always.